### PR TITLE
Serhii/mint

### PR DESCRIPTION
**Pull request by SeHor05**

### CODE REVIEW

File: tvm_assembler/src/simple.rs [hunk 1]
```diff
@@ -837,6 +837,8 @@ impl Engine {
         DIFF_PATCH_BINARY_ZIP                => 0xC7, 0x23
         DIFF_PATCH_BINARYQ                   => 0xC7, 0x24
         DIFF_PATCH_BINARY_ZIPQ               => 0xC7, 0x25
+        MINTECC                              => 0xC7, 0x26
+        CNVRTSHELLQ                          => 0xC7, 0x27
```
Comment (SilkovAlexander) [inline tvm_assembler/src/simple.rs:841]: mb change to MINT_ECC and CONVERT_SHELL_Q

File: tvm_block/src/out_actions.rs [hunk 2]
```diff
@@ -189,6 +208,14 @@ impl Serializable for OutAction {
                 mode.write_to(cell)?;
                 value.write_to(cell)?;
             }
+            OutAction::MintToken { ref value } => {
+                ACTION_MINTECC.write_to(cell)?; //tag
```
Comment (SilkovAlexander) [inline tvm_block/src/out_actions.rs:212]: tag?

File: tvm_executor/src/transaction_executor.rs [hunk 3]
```diff
@@ -784,6 +785,43 @@ pub trait TransactionExecutor {
                     }
                 }
                 OutAction::CopyLeft { .. } => 0,
+                OutAction::MintToken { value } => {
+                    let mut valuecur = CurrencyCollection::new();
```
Comment (SilkovAlexander) [inline tvm_executor/src/transaction_executor.rs:789]: I don't like `valuecur`

File: tvm_executor/src/transaction_executor.rs [hunk 4]
```diff
@@ -784,6 +785,43 @@ pub trait TransactionExecutor {
...
+                    match acc_remaining_balance.add(&valuecur) {
+                        Ok(_) => {
+                            phase.spec_actions += 1;
+                            0
+                        }
+                        Err(_) => RESULT_CODE_INVALID_BALANCE,
+                    }
+                }
+                OutAction::ExchangeShell { value } => {
+                    let mut valuecur = CurrencyCollection::new();
```
Comment (SilkovAlexander) [inline tvm_executor/src/transaction_executor.rs:800]: ^

File: tvm_executor/src/transaction_executor.rs [hunk 5]
```diff
@@ -784,6 +785,43 @@ pub trait TransactionExecutor {
...
+                    valuecur.set_other(2, value as u128)?;
+                    if let Some(a) = acc_remaining_balance.other.get(&2)? {
```
Comment (SilkovAlexander) [inline tvm_executor/src/transaction_executor.rs:802]: pls move 2 to constant

File: tvm_assembler/src/simple.rs [hunk 6]
```diff
@@ -837,6 +837,9 @@ impl Engine {
...
+        CALCBKREWARD                         => 0xC7, 0x29
```
Comment (SilkovAlexander) [inline tvm_assembler/src/simple.rs:842]: ^

File: tvm_executor/src/transaction_executor.rs [hunk 7]
```diff
@@ -784,6 +785,43 @@ pub trait TransactionExecutor {
...
+                            phase.spec_actions += 1;
+                            0
+                        }
+                        Err(_) => RESULT_CODE_INVALID_BALANCE,
+                    }
+                }
+                OutAction::ExchangeShell { value } => {
+                    let mut valuecur = CurrencyCollection::new();
+                    if is_special == true {
+                        valuecur.set_other(2, value as u128)?;
```
Comment (SilkovAlexander) [inline tvm_executor/src/transaction_executor.rs:802]: better store ecc key  as a const

File: tvm_executor/src/transaction_executor.rs [hunk 8]
```diff
@@ -784,6 +785,43 @@ pub trait TransactionExecutor {
...
+                        if let Some(a) = acc_remaining_balance.other.get(&2)? {
+                            if a <= VarUInteger32::from_two_u128(0, value as u128)? {
+                                valuecur.other.set(&2, &a)?;
```
Comment (SilkovAlexander) [inline tvm_executor/src/transaction_executor.rs:805]: ^

**Review by SilkovAlexander — APPROVED** [review]

**Review by elasticLove1 — APPROVED** [review]

**Review by SilkovAlexander — APPROVED** [review]

**Pull request merged**